File: Rvfilter/src/main/java/com/rv/FilterActivity.java
package com.rv;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatTextView;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.rv.rv.filter.FilterAdapter;
import com.rv.rv.RVUtils;

import java.util.LinkedHashMap;

/**
 * by y on 2017/4/24
 */

public class FilterActivity extends AppCompatActivity implements View.OnClickListener {


    private FilterAdapter filterAdapter;
    private AppCompatTextView textView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_filter);
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new GridLayoutManager(recyclerView.getContext(), 3));
        filterAdapter = new FilterAdapter(RVUtils.initFilterData());
        recyclerView.setAdapter(filterAdapter);


        findViewById(R.id.ok).setOnClickListener(this);

        textView = (AppCompatTextView) findViewById(R.id.tv);
    }

    @Override
    public void onClick(View v) {
        LinkedHashMap<String, String> selectBean = filterAdapter.getSelectBean();


        textView.setText(selectBean.toString());
    }
}

File: UtilsSample/RvFilter/src/main/java/sample/util/develop/android/rv/filter/FilterActivity.kt
package sample.util.develop.android.rv.filter


import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.view.View
import kotlinx.android.synthetic.main.rv_activity_filter.*
import sample.util.develop.android.rv.filter.widget.RVUtils
import sample.util.develop.android.rv.filter.widget.filter.FilterAdapter

class FilterActivity : AppCompatActivity(), View.OnClickListener {


    private lateinit var filterAdapter: FilterAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.rv_activity_filter)
        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = GridLayoutManager(recyclerView.context, 3)
        filterAdapter = FilterAdapter(RVUtils.initFilterData())
        recyclerView.adapter = filterAdapter
        ok.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        val selectBean = filterAdapter.selectBean
        tv.text = selectBean.toString()
    }
}
